### PR TITLE
Delete Docker images right after a push

### PR DESCRIPTION
**Pull request by termie**

### CODE REVIEW

File: docker/docker.go [hunk 1]
```diff
@@ -956,6 +956,18 @@ func (s *DockerPushStep) Execute(ctx context.Context, sess *core.Session) (int,
...
 	}
+
+	// Delete the image after
+	defer func() {
+		imageName := fmt.Sprintf("%s:%s", s.repository, s.tags[0])
+		err = client.RemoveImage(imageName)
+		if err != nil {
+			s.logger.WithField("imageName", imageName).WithError(err).Debug("Failed to delete image")
+		} else {
+			s.logger.Println("Deleted image:", imageName)
```
Comment (hatchan) [inline docker/docker.go]: Can this log statement be made consistent with the top one? Ie. have a `WithField` and `.Debug`. (same applies to the other section)

File: docker/docker.go [hunk 2]
```diff
@@ -1006,6 +1023,23 @@ func (s *DockerPushStep) tagAndPush(imageID string, e *core.NormalizedEmitter, c
...
+			defer func() {
+				imageName := fmt.Sprintf("%s:%s", s.repository, tag)
+				err = client.RemoveImage(imageName)
+				if err != nil {
+					s.logger.
+						WithError(err).
+						WithField("imageName", imageName).
+						Warn("Failed to delete image")
+				} else {
+					s.logger.Println("Deleted image:", imageName)
```
Comment (kokaz) [inline docker/docker.go]: You forgot this statement?

File: docker/docker.go [hunk 3]
```diff
@@ -1006,6 +1023,23 @@ func (s *DockerPushStep) tagAndPush(imageID string, e *core.NormalizedEmitter, c
 				return 1, err
 			}
 			s.logger.Println("Pushed container:", s.repository, s.tags)
+
+			// Delete the image after
+			defer func() {
+				imageName := fmt.Sprintf("%s:%s", s.repository, tag)
```
Comment (kokaz) [inline docker/docker.go]: Don't you think this could be a standalone function?

Comment (hatchan) [inline docker/docker.go]: Sounds good

File: docker/docker.go [hunk 4]
```diff
@@ -957,21 +957,7 @@ func (s *DockerPushStep) Execute(ctx context.Context, sess *core.Session) (int,
...
-				WithError(err).
-				WithField("imageName", imageName).
-				Warn("Failed to delete image")
-		} else {
-			s.logger.
-				WithField("imageName", imageName).
-				Debug("Deleted image")
-		}
-	}()
+	defer cleanup(s.logger, client, s.repository, s.tags[0])
```
Comment (kokaz) [inline docker/docker.go]: `cleanup` what? container? image?
`cleanupImage` would be more explicit and clearer to the reader

**Pull request merged**